sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/Configs.java

```diff
@@ -103,7 +103,7 @@ public class Configs {
 
     // Whether to process the response on a different thread
     private static final String SWITCH_OFF_IO_THREAD_FOR_RESPONSE_NAME = "COSMOS.SWITCH_OFF_IO_THREAD_FOR_RESPONSE";
-    private static final boolean DEFAULT_SWITCH_OFF_IO_THREAD_FOR_RESPONSE = false;
+    private static final boolean DEFAULT_SWITCH_OFF_IO_THREAD_FOR_RESPONSE = true;
```

Comment:
I don't know the Java stack well enough to comment on this part.

Reply:
Yeah, same reactor issue we saw couple of weeks back, for which I updated the docs / samples to use `publishOn()` operator. 
Worth reading this section `Use Appropriate Scheduler (Avoid stealing Event loop IO Netty threads)` on our perf guide for java SDK - https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/performance-tips-java-sdk-v4?tabs=api-async

And more interesting info here on the reactor pattern - https://spring.io/blog/2019/12/13/flight-of-the-flux-3-hopping-threads-and-schedulers